vllm/v1/core/scheduler.py

```diff
@@ -139,6 +154,10 @@ def schedule(self) -> "SchedulerOutput":
             num_new_tokens = min(num_new_tokens, token_budget)
             assert num_new_tokens > 0
 
+            # Guided decoding related.
+            if request.use_guided_decoding:
+                guided_decoding_request_ids[request.request_id] = req_index
```

Comment:
This is a bug for two reasons:
1. The request may not be scheduled. Please check out L198
2. In case of chunked prefill, the request may not generate any new output token in this step.

Reply:
this makes sense. for 1 I think we need to move this line down.

For 2) we use this to track mapping between request_id -> index_in_batch iirc to determine the mask to be broadcast to the gpu_runner.

Reply:
good catch! I moved this down to where we know for sure the request is scheduled. (fix #1) 

I'm not sure what the bug is with #2 - let me know if we need to talk about it some more